### ISSUE TITLE
Add Search bar in cluster

### ISSUE THREAD

**Issue by anandmuthu007** [someshkar/covid19india-cluster#81]
Try to add search bar for search state and city in cluster.it will become easy to find. State wise patients details.

**Comment by itguyofficial** [comment]
There is a search bar in the Cluster page which search based on the Patient name alone. Is that ok make it support other search criteria as well like State, patient name etc?

**Comment by sibeshkar** [comment]
@SahajR can you try dealing with this? As cases are ballooning, it's becoming more difficult to zoom into the city/state you want to check. Thanks :)

**Comment by SahajR** [comment]
I can do this. I need to know all the possible inputs and expected behavior for each.

For example, what should happen if I search `City Name` ? If in city mode, I can focus on the city node, but if not, what should be focused on?

**Comment by SahajR** [comment]
@sibeshkar Hey just checking back since I didn't hear from you. We can discuss all the UI requirements for search before going ahead.

**Comment by sibeshkar** [comment]
Sorry just missed this among avalanche of notifs. I think best will be if there's a key of each node which is used to fuzzy search through the nodes and zoom in if there's a match. In search bar, this would look like : "Type city, state or patient number to searh...".

**Comment by SahajR** [comment]
Ok so we need to also notify the user that they need to be in a certain filter for a particular search to work. because the clustering graph changes only after a filter is set initially.

**Comment by sibeshkar** [comment]
"Type city, state or patient number to search...". This text will change based on which filter you apply?

**Comment by SahajR** [comment]
Yes, that's fine, but what happens if the user searches `Delhi` if the graph is in patient mode?

**Comment by sibeshkar** [comment]
Nothing happens (we can red-highlight the input box to indicate no matches), which indicates there are no nodes which match

**Comment by SahajR** [comment]
ok. Will start with this tomorrow morning.

**Comment by sibeshkar** [comment]
Thank you!
